src/notation/view/internal/undoredotoolbarmodel.cpp

```diff
@@ -22,98 +22,96 @@
 
 #include "undoredotoolbarmodel.h"
 
-#include "uicomponents/view/menuitem.h"
+#include "uicomponents/view/toolbaritem.h"
 
 using namespace mu::notation;
 using namespace muse;
 using namespace muse::uicomponents;
 
+static const actions::ActionCode UNDO_ACTION_CODE("undo");
+static const actions::ActionCode REDO_ACTION_CODE("redo");
+
 UndoRedoToolbarModel::UndoRedoToolbarModel(QObject* parent)
-    : QObject(parent), Injectable(iocCtxForQmlObject(this))
+    : AbstractToolBarModel(parent)
 {
 }
 
 void UndoRedoToolbarModel::load()
```

Comment:
Isn't a call to `updateItems` necessary in `load`, to ensure that the names are populated when the toolbar is created while a notation is already open?

Reply:
UndoRedoToolbar is created when the program is initialized, i.e. when the notation isn't open.
so I think it would be unnecessary to call updateItems in load

Reply:
With Undo/Redo history, the names of the toolbar items change on ever undoable event.

Reply:
@rpatters1 That is managed by `subsribeOnUndoStackChanges`.